solidity/ecdsa/test/WalletRegistry.WalletCreation.test.ts

```diff
@@ -2818,103 +3121,244 @@ describe("WalletRegistry - Wallet Creation", async () => {
-              "DKG has not timed out"
+          it("should revert with 'seed awaiting has not timed out' error", async () => {
+            await expect(walletRegistry.notifySeedTimeout()).to.be.revertedWith(
+              "seed awaiting has not timed out"
             )
           })
         })
-      })
+        context("with seed timeout period passed", async () => {
+          before(async () => {
+            await createSnapshot()
 
-      context("with dkg timed out", async () => {
-        before(async () => {
-          await createSnapshot()
+            await mineBlocksTo(
+              requestNewWalletStartBlock + params.dkgSeedTimeout
+            )
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          context("called by a third party", async () => {
+            let tx: ContractTransaction
+
+            before(async () => {
+              await createSnapshot()
+
+              tx = await walletRegistry.connect(thirdParty).notifySeedTimeout()
+            })
+
+            after(async () => {
+              await restoreSnapshot()
+            })
 
-          await mineBlocksTo(startBlock + dkgTimeout)
+            it("should emit DkgSeedTimedOut event", async () => {
+              await expect(tx).to.emit(walletRegistry, "DkgSeedTimedOut")
+            })
+
+            it("should clean dkg data", async () => {
+              await assertDkgResultCleanData(walletRegistry)
+            })
+
+            it("should unlock the sortition pool", async () => {
+              await expect(await sortitionPool.isLocked()).to.be.false
+            })
+          })
         })
 
-        after(async () => {
-          await restoreSnapshot()
+        context("with random relay entry submitted", async () => {
+          before(async () => {
+            await createSnapshot()
+            await submitRelayEntry(walletRegistry, randomBeacon)
+          })
+
+          after(async () => {
+            await restoreSnapshot()
+          })
+
+          it("should revert with 'seed awaiting has not timed out' error", async () => {
+            await expect(walletRegistry.notifySeedTimeout()).to.be.revertedWith(
+              "seed awaiting has not timed out"
+            )
+          })
         })
+      })
+    })
+  })
+
+  describe("notifyDkgTimeout", async () => {
+    context("with initial contract state", async () => {
+      it("should revert with 'DKG has not timed out' error", async () => {
+        await expect(walletRegistry.notifyDkgTimeout()).to.be.revertedWith(
+          "DKG has not timed out"
+        )
+      })
+    })
+
+    context("with new wallet requested", async () => {
+      let requestNewWalletStartBlock: number
+
+      before(async () => {
+        await createSnapshot()
+        const tx = await walletRegistry.connect(walletOwner).requestNewWallet()
+
+        requestNewWalletStartBlock = tx.blockNumber
```

Comment:
We can use `tx.blockNumber` directly instead of `requestNewWalletStartBlock` in the test below. I don't think we need to create a new var for this.

Reply:
`requestNewWalletStartBlock` is used only in one place. Instead of `await mineBlocksTo(requestNewWalletStartBlock + dkgTimeout + 1)` we can do `await mineBlocksTo(tx.blockNumber + dkgTimeout + 1)`. In both cases we extract from tx, the only difference is that we introduce a new var as it is coded now. However, it's a test which is no biggy in terms of the gas costs, so the bottom line it's a personal preference. We certainly can leave as is.

Reply:
Yeah, now I see, I was looking at another test case, where this variable is used twice. But for me, it makes sense for consistency to leave it as is, as we have a similar approach in other tests where we define `requestNewWalletStartBlock`.